packages/tx/src/baseTransaction.ts

```diff
@@ -53,9 +54,22 @@ export abstract class BaseTransaction<TransactionObject> {
     const sB = toBuffer(s === '' ? '0x' : s)
 
     this.nonce = new BN(toBuffer(nonce === '' ? '0x' : nonce))
+
+    if (typeof(gasPrice) === "string") {
```

Comment:
@Ghorbanian what is your reasoning for adding this logic to the base class in Transaction, but for Block (difficulty) you only added it to `header-from-rpc.ts`? i guess we should be consistent. integer strings would only be incoming from rpc, so i feel like we could also handle this there.

thinking about this in a new way, i almost feel like we should just handle base 10 integer strings in `toBuffer` then all this work would automatically be covered everywhere. @holgerd77 would you have any hesitations against this suggestion?

Reply:
Yes, I think this makes a lot of sense to integrate into `toBuffer` directly, this PR here is getting far too extensive for what is targeted to be achieved. 

Since we have `PrefixedHexString` now more strict in util, it should also be possible to cleanly differentiate between the two version of strings (so "0x10" vs "10").

Reply:
(but maybe we should also sleep over it one night and give it some thought if this might have side effects we are atm not anticipating)

Reply:
I second that adding the logic to handle integer strings in `toBuffer` would make more sense, but we would be effectively expanding the inputs supported by the `toBuffer` function:
```
Inputs supported: `Buffer`, `String` (hex-prefixed), `Number`, null/undefined, `BN` and other objects
```

The above becomes:

```
Inputs supported: `Buffer`, `String` (hex-prefixed and integer), `Number`, null/undefined, `BN` and other objects
```

The change would involve something like this:

```
if (typeof v === 'string') {
    if (!isHexString(v)) {
      throw new Error(
        `Cannot convert string to buffer. toBuffer only supports 0x-prefixed hex strings and this string was given: ${v}`
      )
    }
    return Buffer.from(padToEven(stripHexPrefix(v)), 'hex')
  }
```

The above becomes:

```
if (typeof v === 'string') {
    if (!isHexString(v)) {
      try {
        return '0x' + parseInt(input, 10).toString(16)
      } catch {
        const msg = `Cannot convert string to buffer. toBuffer only supports integer or 0x-prefixed hex strings but the given string was: ${v}`
        throw new Error(msg)
      }
    }
    return Buffer.from(padToEven(stripHexPrefix(v)), 'hex')
  }
```

So the question is, is there a case where supporting integer string inputs to the `toBuffer` array may cause unexpected issues? I am having difficulty thinking of any simple cases.

Reply:
thanks for digging into this, i will ask the team for feedback on this idea